lib/std/json/hashmap_test.zig

```diff
@@ -29,6 +32,31 @@ test "parse json hashmap" {
     try testing.expectEqual(@as(i32, 1), parsed.value.map.get("xyz").?.i);
 }
 
+test "parse json hashmap while streaming" {
+    const doc =
+        \\{
+        \\  "abc": {"i": 0, "s": "d"},
+        \\  "xyz": {"i": 1, "s": "w"}
+        \\}
+    ;
+    var stream = std.io.fixedBufferStream(doc);
+    var json_reader = jsonReader(testing.allocator, stream.reader());
+
+    var parsed = try parseFromTokenSource(
+        ArrayHashMap(T),
+        testing.allocator,
+        &json_reader,
+        .{},
+    );
+    defer parsed.deinit();
+    // Deinit our reader to invalidate its buffer
+    json_reader.deinit();
```

Comment:
This will not invalidate the string content though, right? For that you'd need to make a copy of `doc` using `testing.allocator`, and then free the copy at this point. As long as the `[]const u8` strings are pointers into the static data of `doc`, we're not really testing this change.

Reply:
If I'm understanding correctly, the keys are pointing into `json_reader`'s buffer, not into `doc`. You can test this by running this test on the master branch, i.e. applying this PR's `hashmap_test.zig` changes but not applying the `hashmap.zig` changes, It will panic when the keys don't exist. Upon printing `parsed.value.map.keys()`, you'll see them full of `0xaa`.
I ran this test without applying the `hashmap.zig` changes, and added a debug print to print keys() before trying to access them (` std.debug.print("{any}\n", .{parsed.value.map.keys()});`). This is partial output of the result:
```
Test [1982/2549] test.parse json hashmap while streaming... { { 170, 170, 170 }, { 170, 170, 170 } }
thread 299268 panic: attempt to use null value
/home/jrz/program/buildzig/zig/lib/std/json/hashmap_test.zig:57:68: 0x129b7d1 in test.parse json hashmap while streaming (test)
    try testing.expectEqualStrings("d", parsed.value.map.get("abc").?.s);
                                                                   ^
```

Reply:
No worries, thank you for the review :heart: